x-pack/plugins/xpack_main/index.js

```diff
@@ -13,34 +13,26 @@ import {
 import { mirrorPluginStatus } from '../../server/lib/mirror_plugin_status';
 import { replaceInjectedVars } from './server/lib/replace_injected_vars';
 import { setupXPackMain } from './server/lib/setup_xpack_main';
-import { getLocalizationUsageCollector } from './server/lib/get_localization_usage_collector';
-import { createTelemetryUsageCollector } from './server/lib/telemetry';
-import { uiCapabilitiesForFeatures } from './server/lib/ui_capabilities_for_features';
 import {
   xpackInfoRoute,
   featuresRoute,
   settingsRoute,
 } from './server/routes/api/v1';
-import { telemetryRoute } from './server/routes/api/v2';
-import {
-  CONFIG_TELEMETRY,
-  getConfigTelemetryDesc,
-} from './common/constants';
-import mappings from './mappings.json';
 import { i18n } from '@kbn/i18n';
 
-export { callClusterFactory } from './server/lib/call_cluster_factory';
 import { registerOssFeatures } from './server/lib/register_oss_features';
+import { uiCapabilitiesForFeatures } from './server/lib/ui_capabilities_for_features';
+import { has } from 'lodash';
 
-/**
- * Determine if Telemetry is enabled.
- *
- * @param {Object} config Kibana configuration object.
- */
-function isTelemetryEnabled(config) {
-  return config.get('xpack.xpack_main.telemetry.enabled');
+function movedToTelemetry(configPath) {
+  return (settings, log) => {
+    if (has(settings, configPath)) {
+      log(`Config key ${configPath} is deprecated. Configuration can be inferred from the "telemetry" settings`);
```

Comment:
This message is not very useful, as it does not tell the user what the new key is.

Reply:
I tried following what other plugins are doing. I do agree we can refine it